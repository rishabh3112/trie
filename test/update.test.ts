import Store from "../src";

describe("update funtionality", () => {
  it("should update the existing item", async () => {
    const store = new Store<string[]>("pokemon");
    await store.add("ash ketchup", ["pickachu"]);

    const ash = await store.at("ash ketchup");

    await store.update(ash.id, "ash ketchum", ["pickachu", "chalizard"]);

    const ashKetchum = await store.at("ash ketchum");
    expect(ashKetchum).toMatchSnapshot();

    try {
      await store.at("ash ketchup");
    } catch (err) {
      expect(err).toMatchSnapshot();
    }
  });

  it("should update if key is same", async () => {
    const store = new Store<string[]>("pokemon");
    await store.add("ash ketchum", ["pickachu"]);

    const ash = await store.at("ash ketchum");

    await store.update(ash.id, "ash ketchum", ["pickachu", "chalizard"]);

    const ashKetchum = await store.at("ash ketchum");
    expect(ashKetchum).toMatchSnapshot();
  });

  it("should throw error if id doesn't exists", async () => {
    const store = new Store<string[]>();
    try {
      await store.update("id", "ash ketchum", ["pickachu", "chalizard"]);
    } catch (err) {
      expect(err).toMatchSnapshot();
    }
  });
});
