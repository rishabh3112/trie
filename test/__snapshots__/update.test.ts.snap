// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`update funtionality should throw error if id doesn't exists 1`] = `"Data doesn't exists"`;

exports[`update funtionality should update if key is same 1`] = `
Object {
  "data": Array [
    "pickachu",
    "chalizard",
  ],
  "id": "pokemon#11",
  "key": "ash ketchum",
}
`;

exports[`update funtionality should update the existing item 1`] = `
Object {
  "data": Array [
    "pickachu",
    "chalizard",
  ],
  "id": "pokemon#12",
  "key": "ash ketchum",
}
`;

exports[`update funtionality should update the existing item 2`] = `"Data doesn't exists"`;
